### PR TITLE
profile call needs to happen before event trigger

### PR DESCRIPTION
**Pull request by RiteshHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3335

### Change description ###
Make profile call before event trigger 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/shared/components/case-viewer/services/event-trigger.resolver.ts [hunk 1]
```diff
@@ -40,6 +42,7 @@ export class EventTriggerResolver implements Resolve<CaseEventTrigger> {
     if (-1 === EventTriggerResolver.IGNORE_WARNING_VALUES.indexOf(ignoreWarning)) {
       ignoreWarning = 'false';
     }
+    this.profileService.get().subscribe(_ => this.profileNotifier.announceProfile(_));
```
Comment (phillip-whitaker) [inline src/shared/components/case-viewer/services/event-trigger.resolver.ts:45]: If this profile call truly needs to be done before the case loading, I think it has to be pipe'd in. Maybe using a forkJoin with the case service call.

Comment (RiteshHMCTS) [reply src/shared/components/case-viewer/services/event-trigger.resolver.ts]: I have replaced the http calls with forkJoin

**Review by phillip-whitaker — DISMISSED** [review]

**Review by adnanakgun — DISMISSED** [review]

File: src/shared/components/case-viewer/services/event-trigger.resolver.ts [hunk 2]
```diff
@@ -40,10 +42,17 @@ export class EventTriggerResolver implements Resolve<CaseEventTrigger> {
     if (-1 === EventTriggerResolver.IGNORE_WARNING_VALUES.indexOf(ignoreWarning)) {
       ignoreWarning = 'false';
     }
-    return this.casesService
-      .getEventTrigger(caseTypeId, eventTriggerId, cid, ignoreWarning)
+
+    const profileObserver = this.profileService.get();
+    const eventTriggerObserver = this.casesService.getEventTrigger(caseTypeId, eventTriggerId, cid, ignoreWarning);
+
+    return forkJoin([profileObserver, eventTriggerObserver])
```
[the remark this answers is not in the document]
Comment (andywilkinshmcts) [reply src/shared/components/case-viewer/services/event-trigger.resolver.ts:49]: I think you need to put the call to getEventTrigger in the handler for the profileService, something like this (very rough, not compiled or tested but you get the idea)
profileObserver
.pipe(
map(profileDate) => {
  this.profileNotifier.announceProfile(profileData);
  return this.casesService
      .getEventTrigger(caseTypeId, eventTriggerId, cid, ignoreWarning)
      .pipe (
      map (eventTrigger => this.cachedEventTrigger = eventTrigger))

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]

File: src/shared/components/case-viewer/services/event-trigger.resolver.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (RiteshHMCTS) [reply src/shared/components/case-viewer/services/event-trigger.resolver.ts]: made the first call to profiler and then to case service.

**Review by phillip-whitaker — APPROVED** [review]

**Review by kuda1992 — DISMISSED** [review]
looks good

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hmcts/ccd-case-ui-toolkit",
-  "version": "3.1.5-complex-field-check-answers",
+  "version": "3.1.5-profile-call-before-event-trigger",
```
Comment (andywilkinshmcts) [inline package.json:3]: Is this correct? On main the current version  "3.1.5-complex-field-check-answers"
Should we not be incrementing the version

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Check with mo what version number to use

**Review by kuda1992 — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (RiteshHMCTS) [reply package.json:3]: we don't do version increment only postfix ticket reference and the same confirmed with @Mo-Lala84

**Review by andywilkinshmcts — APPROVED** [review]

**Pull request merged**